daemon/containerd/image_commit.go

```diff
@@ -217,28 +227,46 @@ func writeContentsForImage(ctx context.Context, snName string, cs content.Store,
 }
 
 // createDiff creates a layer diff into containerd's content store.
-func createDiff(ctx context.Context, name string, sn snapshots.Snapshotter, cs content.Store, comparer diff.Comparer) (ocispec.Descriptor, digest.Digest, error) {
+// If the diff is empty it returns nil empty digest and no error.
+func createDiff(ctx context.Context, name string, sn snapshots.Snapshotter, cs content.Store, comparer diff.Comparer) (*ocispec.Descriptor, digest.Digest, error) {
 	newDesc, err := rootfs.CreateDiff(ctx, name, sn, comparer)
 	if err != nil {
-		return ocispec.Descriptor{}, "", err
+		return nil, "", err
+	}
+
+	ra, err := cs.ReaderAt(ctx, newDesc)
+	if err != nil {
+		if cerrdefs.IsNotFound(err) {
```

Comment:
What does it mean if we get a not found error here and why aren't we returning it?

Reply:
It shouldn't happen now. I was meaning it to turn into a fast-path if `CreateDiff` behavior would change in future so that it doesn't produce an archive when the diff is empty.

Although, maybe it's better to fail loudly in case there's an actual error with `CreateDiff` and somehow it doesn't produce the archive even when it should?